UI/src/view/LinearAssetLabel.js

```diff
@@ -4,22 +4,47 @@
         AssetLabel.call(this);
         var me = this;
 
-        var backgroundStyle = new ol.style.Style({
+        var backgroundStyle = function (value) {
+
+          var valueLength = value.toString().length;
+
+          if (valueLength > 6 || value < 0) {
+            return new ol.style.Style({
+              image: new ol.style.Icon(({
+                src: 'images/warningLabel.png'
+              }))
+            });
+          }else if (valueLength > 4 && valueLength < 7) {
+            return new ol.style.Style({
+              image: new ol.style.Icon(({
+                src: 'images/linearLabel_background_large.png'
+              }))
+            });
+          }
+          return new ol.style.Style({
```

Comment:
The code before could be simplified with something like that
            switch (true){
                case (valueLength > 6 || value < 0):
                    src_image = 'images/warningLabel.png';
                    break;
                case (valueLength > 4 && valueLength < 7):
                    src_image = 'images/linearLabel_background_large.png';
                    break;
                default:
                    src_image = 'images/linearLabel_background.png';
                    break;
            }